.azuredevops/platformPipelines/platform.apiSpecs.yml

```diff
@@ -0,0 +1,61 @@
+name: '.Platform - Update API Specs file'
+
+pr: none
+
+schedules:
+- cron: "0 12 * * 0"
+  displayName: Weekly Sunday Update
+  branches:
+    include:
+    - main
+
+variables:
+  - template: '../../settings.yml'
+  - name: pipelinePrincipalGitUserName
+    value: 'CARMLPipelinePrincipal'
+  - name: pipelinePrincipalGitUserEmail
+    value: 'CARML@noreply.github.com'
+
+jobs:
+  - job: Update_API_Specs_file
+    pool:
+      ${{ if ne(variables.vmImage, '') }}:
+        vmImage: ${{ variables.vmImage }}
+      ${{ if ne(variables.poolName, '') }}:
+        name: ${{ variables.poolName }}
+    steps:
+      - checkout: self
+        persistCredentials: true
+
+      - task: PowerShell@2
+        displayName: 'Update file'
+        inputs:
+          targetType: inline
+          pwsh: true
+          script: |
+            # Load used functions
+            . (Join-Path '$(System.DefaultWorkingDirectory)' 'utilities' 'tools' 'platform' 'Set-ApiSpecsFile.ps1')
+
+            $functionInput = @{
+              SpecsFilePath = Join-Path '$(System.DefaultWorkingDirectory)' 'utilities' 'src' 'apiSpecsList.json'
+            }
+
+            Write-Verbose "Invoke task with" -Verbose
+            Write-Verbose ($functionInput | ConvertTo-Json | Out-String) -Verbose
+
+            Set-ApiSpecsFile @functionInput -Verbose
+
+      - task: PowerShell@2
+        displayName: 'Push changes'
+        inputs:
+          targetType: inline
+          pwsh: true
+          script: |
+            git config --global user.email '$(pipelinePrincipalGitUserEmail)'
+            git config --global user.name '$(pipelinePrincipalGitUserName)'
+            Write-Verbose '$(Build.SourceBranch)' -Verbose
+            git add .
+            git status
+            git commit -m 'Push updated API Specs file'
+            git pull $(Build.Repository.Uri) HEAD:$(Build.SourceBranch)
```

Comment:
Why having different git cmdlets for ADO and GH?

Reply:
I guess this stemps from pipelines I used as a foundation. Aligned them now. The only 'real' difference was the way the pull & push commands are written.

Reply:
Ok, nevermind. Just undid the change. ADO apparently works a bit different with its checkout logic. By default it runs into a 'detached' state, breaking the logic in the pipeline. Hence this explicit specification of the repo URI & branch. 

In GitHub this issues does not exist